### PR TITLE
[CRYPTO] Adds NetflixVaultGenerator

### PR DESCRIPTION
**Pull request by svdmaar**
Adds Netflix Vault Generator.

### CODE REVIEW

File: Tests/cryptography/NetflixVaultGenerator/main.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+#include <openssl/bio.h>
+#include <openssl/evp.h>
+#include <openssl/aes.h>
+#include <iostream>
+#include <fstream>
+#include <openssl/modes.h>
+
+using namespace std;
+
+const uint32_t g_esnSize = 42;
```
Comment (sebaszm) [inline Tests/cryptography/NetflixVaultGenerator/main.cpp]: Why is the ESN size fixed? I don't think it's always 42...? Did a quick search on the net and found different sizes of sample ESNs. The TA even has dedicated API to retrieve the ESN size, hence my doubts.

File: Tests/cryptography/NetflixVaultGenerator/main.cpp [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    NetflixData netflixData;
+
+    // 1. Create random IV vector and salt.
+    srand(time(nullptr));
+    uint8_t iv[16];
+    for (int i = 0; i < 16; i++) {
+        iv[i] = rand() % 256;
+        netflixData.m_salt[i] = rand() % 256;
+    }
```
Comment (sebaszm) [inline Tests/cryptography/NetflixVaultGenerator/main.cpp]: OpenSSL has cryptography-strong random number generator functions RAND_***, why not use them instead?

File: Tests/cryptography/NetflixVaultGenerator/main.cpp [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+    // 5. Write IV followed by encrypted NetflixData.
+    outFile.write(reinterpret_cast<const char *>(iv), sizeof(iv));
+    outFile.write(reinterpret_cast<const char *>(&encryptBuffer), sizeof(netflixData));
+
+    outFile.close();
+    inFile.close();
+
+    cout << "SUCCESS: Written netflix fault to " << argv[2] << endl;
```
Comment (sebaszm) [inline Tests/cryptography/NetflixVaultGenerator/main.cpp]: Yeah, it's their fault! :)

**Review by sebaszm — CHANGES_REQUESTED** [review]

**Comment by svdmaar** [comment]
@sebaszm , thanks for you review feedback, I agree with all the points raised and amended a commit. Please have another look at this PR. Thanks

**Review by sebaszm — APPROVED** [review]
LGTM!

**Pull request merged**